spec/features/user_login_logout_functionality_spec.rb

```diff
@@ -10,14 +11,69 @@
       fill_in('new_user[password]', with: 'pass')
       fill_in('new_user[password_confirmation]', with: 'pass')
       click_button "Create"
-      expect(current_path).to eq(dashboard_path)
       click_link("Logout")
+    end
+
+    it "logs in the new user properly with valid info submitted" do
       click_link("Login")
-      fill_in("Email", with: "J@gmail.com")
+      fill_in("Email", with: 'J@gmail.com')
       fill_in("Password", with: 'pass')
-      click_button 'Login'
+      click_button "Login"
       expect(current_path).to eq(dashboard_path)
     end
 
+    it "won't allow a logged in user to log in" do
+      click_link("Login")
+      fill_in("Email", with: 'J@gmail.com')
+      fill_in("Password", with: 'pass')
+      click_button "Login"
+      visit '/login'
+      expect(page).to have_content("Please log out Before switching User")
+      expect(page).not_to have_content("Login")
+    end
+  end
+
+  describe "the registered user can't log in with invalid info" do
+    before(:each) do
+      visit "/"
+      click_link("Create Account")
+      fill_in('new_user[email]', with: 'J@gmail.com')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.